app/forms/steps/claim_details_form.rb

```diff
@@ -2,26 +2,49 @@
 
 module Steps
   class ClaimDetailsForm < Steps::BaseFormObject
-    BOOLEAN_FIELDS = %i[supplemental_claim preparation_time].freeze
+    attr_writer :preparation_time, :work_before, :work_after
+
+    BOOLEAN_FIELDS = %i[supplemental_claim].freeze
+
+    VIEW_BOOLEAN_FIELDS = %i[supplemental_claim preparation_time  work_before work_after].freeze
 
     attribute :prosecution_evidence, :string
     attribute :defence_statement, :string
     attribute :number_of_witnesses, :integer
     attribute :time_spent, :time_period
+    attribute :work_before_date, :multiparam_date
+    attribute :work_after_date, :multiparam_date
 
     validates :prosecution_evidence, presence: true
     validates :defence_statement, presence: true
     validates :number_of_witnesses, presence: true, numericality: { only_integer: true, greater_than: 0 }
     validates :time_spent, numericality: { only_integer: true, greater_than: 0 }, time_period: true,
-      if: ->(form) { form.preparation_time == YesNoAnswer::YES }
+      if: ->(form) { @preparation_time == "yes"}
+    validates :work_before_date, presence: true, multiparam_date: { allow_past: true, allow_future: false },
+      if: ->(form) { @work_before == "yes"}
+    validates :work_after_date, presence: true, multiparam_date: { allow_past: true, allow_future: false },
+      if: ->(form) { @work_after == "yes"}
```

Comment:
we should be using the `@` accessor here, that is what the methods below are for. This would result in issues as we use the validation process in the task list to determine if the step is complete.

Additionally we should be using YesNoAnswer::YES over hardcoding the `'yes'` string